dashboard/client/src/components/DataOverviewTable.tsx

```diff
@@ -116,59 +121,146 @@ const DataOverviewTable = ({
-                  <TableCell align="center">
-                    {formatDateFromTimeMs(start_time * 1000)}
-                  </TableCell>
-                  <TableCell align="center">
-                    {end_time && formatDateFromTimeMs(end_time * 1000)}
-                  </TableCell>
-                </TableRow>
-              ),
-            )}
+            {list.map((dataset, index) => (
+              <DatasetTable
+                datasetMetrics={dataset}
+                isExpanded={expanded[dataset.dataset as keyof {}]}
+                setIsExpanded={(isExpanded: boolean) => {
+                  const copy = { ...expanded, [dataset.dataset]: isExpanded };
+                  setExpanded(copy);
+                }}
+                key={dataset.dataset}
+              />
+            ))}
           </TableBody>
         </Table>
       </div>
     </div>
   );
 };
 
+const useStyles = makeStyles((theme) =>
+  createStyles({
+    icon: {
+      width: 16,
+      height: 16,
+    },
+  }),
+);
+
+const DatasetTable = ({
+  datasetMetrics,
+  isExpanded,
+  setIsExpanded,
+}: {
+  datasetMetrics: DatasetMetrics;
+  isExpanded: boolean;
+  setIsExpanded: CallableFunction;
+}) => {
+  const classes = useStyles();
+  const {
+    dataset,
+    state,
+    progress,
+    total,
+    ray_data_current_bytes,
+    ray_data_output_bytes,
+    ray_data_spilled_bytes,
+    start_time,
+    end_time,
+    operators,
+  } = datasetMetrics;
+  const operatorRows =
+    isExpanded &&
+    operators.map((operator) => (
+      <TableRow key={operator.operator}>
+        <TableCell></TableCell>
+        <TableCell align="left">{operator.operator}</TableCell>
+        <TableCell align="right" size={"small"}>
+          <TaskProgressBar
+            showLegend={false}
+            numFinished={operator.progress}
+            numRunning={
+              operator.state === "RUNNING"
+                ? operator.total - operator.progress
+                : undefined
+            }
+            numCancelled={
+              operator.state === "FAILED"
+                ? operator.total - operator.progress
+                : undefined
+            }
+            total={operator.total}
+          />
+        </TableCell>
+        <TableCell align="center">
+          <StatusChip type="task" status={operator.state} />
+        </TableCell>
+        <TableCell align="right">
+          {memoryConverter(Number(operator.ray_data_output_bytes.max))}
+        </TableCell>
+        <TableCell align="right">
+          {memoryConverter(Number(operator.ray_data_current_bytes.value))}/
+          {memoryConverter(Number(operator.ray_data_current_bytes.max))}
+        </TableCell>
+        <TableCell align="right">
+          {memoryConverter(Number(operator.ray_data_spilled_bytes.max))}
+        </TableCell>
+        <TableCell></TableCell>
+        <TableCell></TableCell>
+      </TableRow>
+    ));
+  return (
+    <React.Fragment>
+      <TableRow key={dataset}>
```

Comment:
key should not be set here. It's only necessary when used directly as an element in a list.